dxr/plugins/clang/menus.py

```diff
@@ -118,31 +118,41 @@ def _more_menu_items(self, (macro_name,)):
 class TypeRef(_QualnameRef):
     @classmethod
     def _condensed_menu_data(cls, tree, prop):
-        """Hang onto qualname *and* kind."""
         return (super(TypeRef, cls)._condensed_menu_data(tree, prop) +
-                (prop.get('kind', ''),))
+                (prop.get('kind', 'type'), 'has_subclass' in prop, 'has_base_class' in prop))
 
-    def _more_menu_items(self, (qualname, kind)):
+    def _more_menu_items(self, (qualname, kind, has_subclass, has_base_class)):
         """Return menu for type reference."""
+        def kind_plural():
```

Comment:
This is fine for now. Any more branches, and we'd probably want to switch to a dict and `.get(kind, 'types')`.